torch/utils/_sympy/value_ranges.py

```diff
@@ -530,12 +530,13 @@ def __getattr__(self, name):
         return self.default_handler
 
 
-def bound_sympy(expr: sympy.Expr, ranges: Dict[sympy.Symbol, ValueRanges]) -> ValueRanges:
-    # Add dynamic shapes within the expression as potentially unbounded
-    dynamic_shapes = expr.free_symbols - ranges.keys()
-    if dynamic_shapes:
-        ranges = deepcopy(ranges)
-        for s in dynamic_shapes:
-            ranges[s] = ValueRanges(0, math.inf)  # type: ignore[index]
-
-    return sympy_interp(SymPyValueRangeAnalysis(), ranges, expr)
+def bound_sympy(expr: sympy.Expr, ranges: Dict[sympy.Symbol, ValueRanges], unbounded_symbols=True) -> ValueRanges:
+    if unbounded_symbols:
+        # Add dynamic shapes within the expression as potentially unbounded
+        dynamic_shapes = expr.free_symbols - ranges.keys()
+        if dynamic_shapes:
+            ranges = deepcopy(ranges)
+            for s in dynamic_shapes:
+                ranges[s] = ValueRanges(0, math.inf)  # type: ignore[index]
```

Comment:
This doesn't look valid; symbols are not necessarily positive.

Reply:
This is pretty much following your patch from https://github.com/pytorch/pytorch/pull/103470, in particular the bit
```python
        for k, v in indices_ranges.items():
            if free_symbols(v):
                v = math.inf
            self.replace_indirect(k, ValueRanges(0, v))
```
What would be the correct thing of doing here?

Reply:
So, I see we wrap constants in `wrap_unspecialized_primitive`, which may create a non-positive `sXX` symbol. Wouldn't it be better to prefix these constants with another letter (I see `iXX` is used, why not use that one?) and leave the `s` for size-like symbols (i.e. integral and bounded below by 2)?